pkg/env/validate.go

```diff
@@ -0,0 +1,82 @@
+package env
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os/exec"
+
+	"github.com/coreeng/corectl/pkg/gcp"
+	"github.com/coreeng/developer-platform/pkg/environment"
+)
+
+var (
+	ErrCloudPlatformNotSupported = errors.New("cloud platform is not supported")
+	ErrInvalidEnvironment        = errors.New("environment is not valid")
+)
+
+// Validate checks if the required tools and configurations for the environment are installed and set up correctly.
+func Validate(ctx context.Context, env *environment.Environment, cmd Commander, client *gcp.Client) error {
+	if env == nil {
+		return ErrInvalidEnvironment
+	}
+
+	if err := depsInstalled(cmd); err != nil {
+		return err
+	}
+
+	if err := platform(env); err != nil {
+		return err
+	}
+
+	if err := cluster(ctx, client, env); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func platform(env *environment.Environment) error {
+	if env.Platform.Type() != environment.GCPVendorType {
+		return ErrCloudPlatformNotSupported
+	}
+	return nil
+}
+
+type Commander interface {
+	CommandOutput(string, ...string) ([]byte, error)
+}
+
+type Command struct {
+}
+
+func (c *Command) CommandOutput(cmd string, args ...string) ([]byte, error) {
```

Comment:
I don't see much profit in having this wrapper function. Or it is just for mocking purposes?